File: core/src/main/java/javabot/operations/time/TimezonesChile.java
package javabot.operations.time;

public class TimezonesChile implements Timezones {

    private static Tri<String> timezones = new Tri<String>( );

    static {
        for( Timezones timezone: Timezones.values( ) ) {
            timezones.insert( timezone.name( ), timezone.getTimezone( ) );
        }
    }

    private enum Timezones {
        Noronha( "Pacific/Easter" ),
        Santiago( "America/Santiago" );

        private String timezone;

        Timezones( String timezone ) {
            this.timezone = timezone;
        }

        public String getTimezone( ) {
            return timezone;
        }
    }

    @Override
    public String get( String province ) {
        return timezones.get( province );
    }

    @Override
    public String getCapital( ) {
        return Timezones.Santiago.getTimezone( );
    }
}

File: src/main/kotlin/javabot/operations/time/TimezonesChile.kt
package javabot.operations.time

class TimezonesChile : Timezones {

    private enum class Timezones(val timezone: String) {
        Noronha("Pacific/Easter"),
        Santiago("America/Santiago"),
    }

    override fun get(province: String): String? {
        return timezones.get(province)
    }

    override fun getCapital(): String {
        return Timezones.Santiago.timezone
    }

    companion object {

        private val timezones = Tri<String>()

        init {
            for (timezone in Timezones.values()) {
                timezones.insert(timezone.name, timezone.timezone)
            }
        }
    }
}
